docker-mac.sh

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+
+def=$'\e[39m'
+yel=$'\e[93m'
+blue=$'\e[34m'
+green=$'\e[32m'
+cyan=$'\e[96m'
+
+# Install Brew  
+echo $blue"==> brew checking"
+BREW_VERSION="$(brew --version 2>/dev/null)"
+echo $def${BREW_VERSION}
+if [[ "$BREW_VERSION" == *"Homebrew"* ]]; then
+  echo $green"brew is installed"
+else
+  echo $yel"brew does not exist"
+  echo $def"please read this link https://brew.sh/ for the information about brew"
+  echo $cyan"Installing homebrew"
+  /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+fi
+
+# Install Docker-Desktop on Mac
+DOCKER_INFO="$(docker info 2>/dev/null | grep ' Operating System: Docker Desktop')"
+echo $def"${DOCKER_INFO}"
+if [[ "$DOCKER_INFO" == *"Operating System: Docker Desktop"* ]]; then
+  echo $green"Docker Desktop is installed"
+else
+  echo $green"Installing Docker Desktop on Mac"$def
+  brew cask install Docker
+  echo $green"Open Docker.app"
+  open /Applications/Docker.app
+  echo $cyan"Docker Desktop needs privileged access"
+  until docker info 2>/dev/null | grep "Server Version:";
+  do
+  echo $yel"Please Grant privilege access and waiting docker to start "$def;
+  sleep 4; 
+  done && echo "Docker is Ready"
+  until kubectl version --short 2>/dev/null | grep "Server Version: v";
+  do
+  echo $cyan"Please Enable Kubernetes in Preferences Menu and waiting Kubernetes to start"$def;
+  sleep 10;
+  done && echo "Kubernetes is Ready"
+fi
+
+# Apply service account, role-binding, namespace, configmap
+echo $green"Apply service-account, role-binding"$def
+kubectl apply -f base/service-account.yaml -f base/role-binding.yaml -f base/namespace.yaml -f base/configmap.yaml
+
+# install helm
+echo $blue"==> Helm checking"
+HELM_VERSION="$(helm version --short 2>/dev/null)"
+echo $def${HELM_VERSION}
+if [[ "$HELM_VERSION" == *"Client: v"* ]]; then
+  echo $green"helm is installed"
+else
+  echo $yel"helm does not exist"
+  echo $cyan"Installing helm"
+  brew install kubernetes-helm
+  echo $green"helm already installed" 
+fi
+
+# Deploy tiller
+echo $green"Deploy tiller"$def
+helm init --service-account tiller --wait
+
+# Install prometheus with helm
+echo $green"Install prometheus with helm"$def
+sudo helm repo update
+sudo helm install stable/prometheus --namespace monitoring --name prometheus
```

Comment:
Does the default install of prometheus include config to scrape pods with `prometheus.io/` annotations?

Reply:
you mean this one 
![image](https://user-images.githubusercontent.com/54768104/67185327-247bdb00-f418-11e9-85b3-3d5a7c19b338.png)

Reply:
No not that one, that's actually a label modifier for after the service has already been discovered by prometheus. I'm talking about the SD config in `prometheus.yaml`. It'll probably be in a configmap.
Have a look at/play with some prometheus configs and see how prometheus metric labelling works - that's where these annos are used.

Also, the above is incorrect, for 2 reasons:
1. Prometheus already scrapes itself internally so it doesn't need to use SD to do it.
2. That's an annotation on the _service_, which is not what we want. Granted, you will see it done a lot around the internets, but it's wrong because when you scrape a service you load balance the scrapes among the available pods. This will lead to incomplete or biased or duplicated metrics. These prom annotations **must** go on _pod_ template annotations.

Reply:
default prometheus configs already have :
- job_name: kubernetes-apiservers
 - role: endpoints
- job_name: kubernetes-nodes
 - role: node
- job_name: kubernetes-nodes-cadvisor
 - role: node
- job_name: kubernetes-service-endpoints
 - role: endpoints
- job_name: prometheus-pushgateway
 - role: service
- job_name: kubernetes-services
 - role: service
- job_name: kubernetes-pods
 - role: pod